client/src/components/related_items/Modal.jsx

```diff
@@ -1,26 +1,31 @@
 import React, {useState, useRef, useEffect} from 'react';
 import {faXmark} from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { isDOMComponent } from 'react-dom/test-utils';
 
 
 const Modal = ({showModal, closeModal, children})=>{
 
     const modalRef = useRef(null);
 
     const clickedOutside = (e) => {
-
+        if(showModal && modalRef.current && !modalRef.current.contains(e.target)) {
+            closeModal();
+        }
     }
     
-    
     useEffect(()=>{
-
+        document.addEventListener('mousedown', clickedOutside);
+        return ()=> {
+            document.removeEventListener('mousedown', clickedOutside);
+        }
     },[showModal])
```

Comment:
I like the way you set up event listener with useEffect and return callback function